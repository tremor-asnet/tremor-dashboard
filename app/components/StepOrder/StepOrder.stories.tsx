import type { Meta, StoryObj } from "@storybook/react";

// Components
import StepOrder from ".";

// Icons
import { IoMdCheckmark } from "react-icons/io";

const meta = {
  title: "Components/StepOrder",
  component: StepOrder,
  tags: ["autodocs"],
} as Meta<typeof StepOrder>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Primary: Story = {
  render: () => (
    <StepOrder
      backgroundColor="bg-few"
      iconInfo={<IoMdCheckmark />}
      titleInfo="Order delivered"
    />
  ),
};
