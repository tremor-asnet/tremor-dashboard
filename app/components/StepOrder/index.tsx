import { StepOrderProps } from "@/types/orderDetail";

const StepOrder = ({
  iconInfo,
  titleInfo,
  descriptionInfo,
  className,
  backgroundColor,
}: StepOrderProps) => (
  <li className={`relative h-fit ${className}`}>
    <div className="flex justify-start px-6 pt-6 leading-[1.3rem] after:content-['']">
      <span
        className={`mr-3 flex h-8 w-8 items-center justify-center rounded-full ${backgroundColor} text-sm font-medium text-dark-primary`}>
        {iconInfo}
      </span>
      <div className="flex flex-col">
        <span className="text-tremor-default text-primary font-semibold after:absolute after:flex after:text-[0.8rem] after:content-[data-content] dark:text-dark-primary">
          {titleInfo}
        </span>
        <span className="text-tremor-label text-primary font-light dark:text-dark-romance">
          {descriptionInfo}
        </span>
      </div>
    </div>
  </li>
);

export default StepOrder;
