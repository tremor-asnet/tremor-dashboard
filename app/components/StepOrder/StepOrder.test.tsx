import { render } from "@testing-library/react";

// Components
import StepOrder from "../StepOrder";

// Icons
import { IoMdCheckmark } from "react-icons/io";

const StepOrderProps = {
  backgroundColor: "bg-few",
  iconInfo: <IoMdCheckmark />,
  titleInfo: "Order delivered",
};

const StepOrderComponent = () => render(<StepOrder {...StepOrderProps} />);

describe("StepOrder Component", () => {
  test("render StepOrder component with snapshot correctly", () => {
    const { container } = StepOrderComponent();

    expect(container).toMatchSnapshot();
  });
});
