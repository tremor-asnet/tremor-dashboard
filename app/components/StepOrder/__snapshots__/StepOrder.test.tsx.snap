// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepOrder Component render StepOrder component with snapshot correctly 1`] = `
<div>
  <li
    class="relative h-fit undefined"
  >
    <div
      class="flex justify-start px-6 pt-6 leading-[1.3rem] after:content-['']"
    >
      <span
        class="mr-3 flex h-8 w-8 items-center justify-center rounded-full bg-few text-sm font-medium text-dark-primary"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M186.301 339.893L96 249.461l-32 30.507L186.301 402 448 140.506 416 110z"
          />
        </svg>
      </span>
      <div
        class="flex flex-col"
      >
        <span
          class="text-tremor-default text-primary font-semibold after:absolute after:flex after:text-[0.8rem] after:content-[data-content] dark:text-dark-primary"
        >
          Order delivered
        </span>
        <span
          class="text-tremor-label text-primary font-light dark:text-dark-romance"
        />
      </div>
    </div>
  </li>
</div>
`;
